apps/server/src/modules/common-cartridge/service/common-cartridge-export.service.spec.ts

```diff
@@ -1,28 +1,335 @@
+import { RichTextElementContentDto } from '../common-cartridge-client/card-client/dto/rich-text-element-content.dto';
+import { LinkElementResponseDto } from '../common-cartridge-client/card-client/dto/link-element-response.dto';
+import { LinkElementContentDto } from '../common-cartridge-client/card-client/dto/link-element-content.dto';
+import { ComponentTextPropsDto } from '../common-cartridge-client/lesson-client/dto/component-text-props.dto';
+import { BoardLessonDto } from '../common-cartridge-client/room-client/dto/board-lesson.dto';
+import { ComponentGeogebraPropsDto } from '../common-cartridge-client/lesson-client/dto/component-geogebra-props.dto';
+import { ComponentLernstorePropsDto } from '../common-cartridge-client/lesson-client/dto/component-lernstore-props.dto';
 
 describe('CommonCartridgeExportService', () => {
 	let module: TestingModule;
 	let sut: CommonCartridgeExportService;
-	let filesStorageServiceMock: DeepMocked<FilesStorageClientAdapterService>;
 	let coursesClientAdapterMock: DeepMocked<CoursesClientAdapter>;
 	let courseRoomsClientAdapterMock: DeepMocked<CourseRoomsClientAdapter>;
 	let cardClientAdapterMock: DeepMocked<CardClientAdapter>;
+	let boardClientAdapterMock: DeepMocked<BoardClientAdapter>;
+	let lessonClientAdapterMock: DeepMocked<LessonClientAdapter>;
+
+	const dummyCourseId = faker.string.uuid();
+	const createXmlString = (nodeName: string, value: boolean | number | string): string =>
+		`<${nodeName}>${value.toString()}</${nodeName}>`;
+	const getFileContent = (archive: AdmZip, filePath: string): string | undefined =>
+		archive.getEntry(filePath)?.getData().toString();
+
+	const setupParams = async (
+		version: CommonCartridgeVersion,
+		exportTopics: boolean,
+		exportTasks: boolean,
+		exportColumnBoards: boolean
+	) => {
+		const courseMetadata = new CourseCommonCartridgeMetadataDto({
+			id: dummyCourseId,
+			courseName: 'TEST COURSE',
+			creationDate: faker.date.recent().toISOString(),
+			copyRightOwners: [faker.person.fullName()],
+		});
+
+		const lessonLinkedTask: LessonLinkedTaskDto[] = [
+			{
+				name: 'First linked Task',
+				description: faker.lorem.paragraph(),
+				descriptionInputFormat: 'plainText',
+				availableDate: faker.date.recent().toISOString(),
+				dueDate: faker.date.future().toISOString(),
+				private: false,
+				publicSubmissions: false,
+				teamSubmissions: false,
+				creator: faker.internet.email(),
+				courseId: dummyCourseId,
+				submissionIds: [],
+				finishedIds: [],
+			},
+			{
+				name: 'second linked Task',
+				description: faker.lorem.paragraph(),
+				descriptionInputFormat: 'plainText',
+				availableDate: faker.date.recent().toISOString(),
+				dueDate: faker.date.future().toISOString(),
+				private: false,
+				publicSubmissions: false,
+				teamSubmissions: false,
+				creator: faker.internet.email(),
+				courseId: dummyCourseId,
+				submissionIds: [],
+				finishedIds: [],
+			},
+		];
+

```

Comment:
Test factories for this dto and also the other dtos used in this test?

Reply:
Done ✅